elastic/beats
# Add container ECS fields in kubernetes metadata

## What does this PR do?
This PR adds `container.id`, `container.runtime` and `container.image.name` in kubernetes metadata to be compliant with https://www.elastic.co/guide/en/ecs/current/ecs-container.html

## Why is it important?
In final events the metadata do not include `container.id`, `container.runtime` and `container.image.name`. 


## Related issues

- Closes https://github.com/elastic/beats/issues/20982
- Related https://github.com/elastic/beats/issues/10926


## How to test this PR
1. Configure and run Filebeat in k8s:
```
filebeat.autodiscover:
      providers:
        - type: kubernetes
          node: ${NODE_NAME}
          hints.enabled: true
          hints.default_config:
            type: container
            paths:
              - /var/log/containers/*${data.kubernetes.container.id}.log
```
Make sure that `container.id`, `container.runtime` and `container.image.name` exist in the final documents in ES.

2. Configure and run Filebeat in k8s:
```
filebeat.inputs:
    - type: container
      paths:
        - /var/log/containers/*.log
      processors:
        - add_kubernetes_metadata:
            host: ${NODE_NAME}
            indexers:
            - container:
            matchers:
            - logs_path:
                logs_path: "/var/log/containers/"
```

## Screenshots
![Screenshot 2020-09-08 at 13 56 03](https://user-images.githubusercontent.com/11754898/92467964-11622d00-f1db-11ea-886b-e650bb9ba20a.png)
